File: src/java/magma/app/Value.java
package magma.app;

import magma.api.Type;
import magma.api.option.Option;
import magma.app.compile.Argument;
import magma.app.compile.Caller;
import magma.app.compile.CompileState;

public interface Value extends Argument, Caller {
    Type resolve(CompileState state);

    Option<String> generateAsEnumValue(String structureName);
}

File: src/node/magma/app/Value.ts
import { Type } from "../../magma/api/Type";
import { CompileState } from "../../magma/app/compile/CompileState";
import { Option } from "../../magma/api/option/Option";
export interface Value extends Argument, Caller  {
	resolve(state: CompileState): Type;
	generateAsEnumValue(structureName: string): Option<string>;
}
